src/sagemaker/huggingface/model.py

```diff
@@ -0,0 +1,308 @@
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Placeholder docstring"""
+from __future__ import absolute_import
+
+import logging
+
+import sagemaker
+from sagemaker import image_uris
+from sagemaker.deserializers import NumpyDeserializer
+from sagemaker.fw_utils import (
+    model_code_key_prefix,
+    validate_version_or_image_args,
+)
+from sagemaker.model import FrameworkModel, MODEL_SERVER_WORKERS_PARAM_NAME
+from sagemaker.predictor import Predictor
+from sagemaker.serializers import NumpySerializer
+
+logger = logging.getLogger("sagemaker")
+
+
+class HuggingFacePredictor(Predictor):
+    """A Predictor for inference against HuggingFace Endpoints.
+
+    This is able to serialize Python lists, dictionaries, and numpy arrays to
+    multidimensional tensors for HuggingFace inference.
+    """
+
+    def __init__(
+        self,
+        endpoint_name,
+        sagemaker_session=None,
+        serializer=NumpySerializer(),
+        deserializer=NumpyDeserializer(),
+    ):
+        """Initialize an ``HuggingFacePredictor``.
+
+        Args:
+            endpoint_name (str): The name of the endpoint to perform inference
+                on.
+            sagemaker_session (sagemaker.session.Session): Session object which
+                manages interactions with Amazon SageMaker APIs and any other
+                AWS services needed. If not specified, the estimator creates one
+                using the default AWS configuration chain.
+            serializer (sagemaker.serializers.BaseSerializer): Optional. Default
+                serializes input data to .npy format. Handles lists and numpy
+                arrays.
+            deserializer (sagemaker.deserializers.BaseDeserializer): Optional.
+                Default parses the response from .npy format to numpy array.
+        """
+        super(HuggingFacePredictor, self).__init__(
+            endpoint_name,
+            sagemaker_session,
+            serializer=serializer,
+            deserializer=deserializer,
+        )
+
+
+def _validate_pt_tf_versions(pytorch_version, tensorflow_version, image_uri):
+
+    if image_uri is not None:
+        return
+
+    if tensorflow_version is not None and pytorch_version is not None:
+        raise ValueError(
+            "tensorflow_version and pytorch_version are both not None. "
+            "Specify only tensorflow_version or pytorch_version."
+        )
+    if tensorflow_version is None and pytorch_version is None:
+        raise ValueError(
+            "tensorflow_version and pytorch_version are both None. "
+            "Specify either tensorflow_version or pytorch_version."
+        )
+
+
+class HuggingFaceModel(FrameworkModel):
+    """An PyTorch SageMaker ``Model`` that can be deployed to a SageMaker ``Endpoint``."""
+
+    _framework_name = "huggingface"
+
+    def __init__(
+        self,
+        model_data,
```

Comment:
could this be optional, since we have the option to create an endpoint with `model_data`.